### PR TITLE
#363: Bump version to 1.5.0 ahead of release

### LINKED ISSUE #363: Bump version to 1.5.0 ahead of release
We will need to update the `VERSION`, `LICENSE`, and top-level `CMakeLists.txt`.

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #363 

This PR:
- bumps version number to 1.5.0 everywhere
- updates `checkpoint` --> `magistrate` in the license
- fixes a problem in the CI where `docker-compose` command was not found

### CODE REVIEW

**Comment by nlslatt** [comment]
CI is broken. Is this new?

File: scripts/license-template
Comment (nlslatt) [inline scripts/license-template:1]: All the files based on this still say checkpoint...

**Comment by cwschilly** [comment]
> CI is broken. Is this new?

No idea. ~~Could it be related to the mismatched licenses? (e.g. [this pipeline](https://github.com/DARMA-tasking/magistrate/actions/runs/10910920434/job/30282438073?pr=364))~~

File: scripts/license-template
Comment (cwschilly) [reply scripts/license-template:1]: Fixed

**Comment by cwschilly** [comment]
I'm not super familiar with magistrate's CI, but [this fix](https://github.com/orgs/community/discussions/116610) seems to work.

It is strange though that the discussion I linked to suggests that we should have run into this problem in April, and it hasn't shown up until now (CI was passing as recently as July).

**Comment by cwschilly** [comment]
@nlslatt @lifflander

`magistrate` pulls in `vt` on one of the CI pipelines. Right now, it targets `develop`--should we change this to `1.5.0` (once vt has released)?

**Review by lifflander — APPROVED** [review]

**Pull request merged**